lib/whitespace.coffee

```diff
@@ -88,3 +93,19 @@ class Whitespace
 
     buffer.transact ->
       buffer.scan new RegExp(spacesText, 'g'), ({replace}) -> replace('\t')
+
+  # returns null if no current Git-repo or file is new and unsaved
+  # returns Array[Number] where each index is a modified row
+  _getModifiedRows: (editor) ->
+    if path = editor?.getPath()
+      if diffs = atom.project.getRepositories()[0]?.getLineDiffs(path, editor.getText())
+        modifiedRows = diffs.map ({oldStart, newStart, oldLines, newLines}) ->
+          startRow = newStart - 1
+          endRow = newStart + newLines - 2
+          # removed section
+          if newLines is 0 and oldLines > 0
```

Comment:
Is `oldLines` ever 0 when `newLines` is 0? I would suspect that shouldn't happen. It would probably be good to verify with some tests and then remove `oldLines` from the condition.